les/handler.go

```diff
@@ -1119,14 +1119,21 @@ func (pm *ProtocolManager) handleMsg(p *peer) error {
 		p.Log().Trace("Received tx status response")
 		var resp struct {
 			ReqID, BV uint64
-			Status    []txStatus
+			Status    []light.TxStatus
 		}
 		if err := msg.Decode(&resp); err != nil {
 			return errResp(ErrDecode, "msg %v: %v", msg, err)
 		}
 
 		p.fcServer.GotReply(resp.ReqID, resp.BV)
 
+		p.Log().Trace("Received helper trie proof response")
+		deliverMsg = &Msg{
+			MsgType: MsgTxStatus,
+			ReqID:   resp.ReqID,
+			Obj:     resp.Status,
+		}
```

Comment:
`SendTxV2Msg` will also respond a `TxStatusMsg`. Unfortunately the `SendTxV2Msg` is not sent by `retriever`. So that the delivered message will cause a peer error.

Reply:
Thanks for finding this one, I missed it because when I wrote this code the tx relay still used the old LES/1 SendTx which did not receive a reply. Fixed, now the tx relay uses the retriever too, with an empty validation function so it does not care about the contents of the reply.
Btw the reply could be used meaningfully to report when the tx is not accepted to the server's tx pool but that requires rewriting the light tx pool which should happen in a later release. Now the operation is identical to the way it has worked until now.